core/server/api/settings.js

```diff
@@ -103,6 +103,15 @@ canEditAllSettings = function (settingsInfo, options) {
                 ));
             }
 
+            if (setting.key === 'activeTheme') {
```

Comment:
The logic is different. Trying to set the activeTheme via the Settings API is a bad request, nothing to do with permissions.  That's why it happens first.

The other if statement is to do with if you do or don't have permission - and permits the changes for internal requests.

Setting activeTheme this way must never be allowed.